### ISSUE TITLE
Can .ogg files be removed?

### ISSUE THREAD

**Issue by jbphet** [phetsims/vibe#19]
When this library was originally created, it required that sounds be supplied in both mp3 and ogg format because Firefox didn't support mp3 and some of the other browsers didn't support ogg.  I just checked, and it looks like support for mp3 files was added to Firefox in release 22, which came out June 25, 2013.  It may be possible to simplify things now and only require one file per sound.

**Comment by samreid** [comment]
@pixelzoom @jbphet and I decided to reopen this discussion now that we are working more heavily on sonification.

**Comment by pixelzoom** [comment]
Publish a dev version without .ogg and have QA test on all platforms.

**Comment by jbphet** [comment]
Note to self: To avoid creating any potential instability on master, I'll be doing a lot of the testing and work for this on branches.  These branches will need to be merged and deleted before this issue is fully addressed.  The branch name with be the same in all cases: `test-off-file-removal`.

- [ ] : Make sure all `test-ogg-file-removal` branches have been merged, deleted, and their associated issues closed.

**Comment by jbphet** [comment]
Data point: I built the Balancing Act simulation before and after the removal of the .ogg files from `GameAudioPlayer`.  I measured the size in each case using the shell command `du -shb`.  Before removal, the resulting sim was 2454795 bytes, after removal it was 2368336 bytes, so the size was reduced by 86459 bytes, which is about 3.7%.  There are four sounds used by the `GameAudioPlayer` type.

**Comment by pixelzoom** [comment]
@jbphet said:
> The branch name with be the same in all cases: `test-off-file-removal`.

Did you mean `test-ogg-file-removal` ?

**Comment by pixelzoom** [comment]
3.7% is a pretty small size decrease.  But I think it will be valuable not to have to create/maintain .ogg files, if they're not needed.

**Comment by pixelzoom** [comment]
@jbphet said:
> I measured the size in each case using the shell command `du -shb` ...

```
% du -shb equality-explorer_all_phet.html 
du: illegal option -- b
usage: du [-H | -L | -P] [-a | -s | -d depth] [-c] [-h | -k | -m | -g] [-x] [-I mask] [file ...]
% 
``

**Comment by samreid** [comment]
86kb of savings sounds great to me.

**Comment by jbphet** [comment]
@pixelzoom said:
> Did you mean `test-ogg-file-removal`?
Yes, and thanks for the edit.

...and also:
> Is -b a Windows-specific option?
I don't think so, I got it off of a man page here: http://linuxcommand.org/lc3_man_pages/du1.html.  It just specifies to output bytes, because the default wasn't giving me the resolution I wanted.

**Comment by pixelzoom** [comment]
> Yes, and thanks for the edit.

I didn't edit.  It still appears 2 different ways in https://github.com/phetsims/vibe/issues/19#issuecomment-370615231.

**Comment by jbphet** [comment]
Fixed.

**Comment by pixelzoom** [comment]
Re `du -b`, @jbphet said:
> If you did it on a Mac, it may not be supported by OSX (which is a BSD derivative I believe).

Yup, that was it.  I didn't realize how much `du` differed across UNIX variants, see https://en.wikipedia.org/wiki/Du_(Unix).  For future reference (writing shell scripts, etc.), `-b` is not part of the SUS (Single UNIX Specification).

**Comment by jbphet** [comment]
I've built two sims with .ogg files removed, one where the files were removed from the sim itself (Ohm's Law), and one where the .ogg files were removed from common code (Balancing Act).

@phet-steele - please test that the audio works on these two simulations on all of our supported platforms.  The .ogg files were originally included to support Firefox, so please be extra vigilant when testing on that browser.

Balancing Act has sound only on the game screen.  Ohm's law has sound effects for the addition and removal of the batteries.

Links:
https://www.colorado.edu/physics/phet/dev/html/ohms-law/1.4.0-testOggFileRemoval.1/phet/ohms-law_en_phet.html
https://www.colorado.edu/physics/phet/dev/html/balancing-act/1.2.0-dev.3/phet/balancing-act_en_phet.html

**Comment by jbphet** [comment]
When testing is compete on this, please assign back to me.

**Comment by phet-steele** [comment]
@jbphet this worked well on all our platforms except IE11. I believe you mentioned it was because you were "creating a new AudioContext", which is invalid in IE11.

**Comment by jbphet** [comment]
I fixed the issue that was causing a problem on IE11, and I'm confident that this won't affect the testing results, since it's an isolated change that doesn't have anything directly to do with the .ogg files.  I'm going to move forward with the removal of the .ogg files and the support thereof.

**Comment by jbphet** [comment]
All .ogg files have been removed, as has all of the code that referenced the .ogg files, mostly indirectly by specifying the name of the audio file without the file type (i.e. `soundEffect` instead of `soundEffect.mp3`).  The license files have also been updated to remove .ogg file entries.

There is one more major piece of work related to this: removing support for the .ogg file format from the audio plugin.  I've logged a separate issue for this in chipper, which is where the audio plugin resides.

I believe that the work is done for removal of the .ogg files, but it would be good to have another round of testing of audio on all supported platforms.  I've discussed this with @phet-steele and he said I should assign the issue to him.  The idea would be to test sound on several simulations from master (i.e. using phet-test) on all supported platforms and make sure it still works.
